extensions/amp-story/1.0/amp-story.js

```diff
@@ -1489,15 +1489,24 @@ export class AmpStory extends AMP.BaseElement {
    * @param {!Object} size including new width and height
    * @private
    */
-  polyfillDvh_(size) {
+  polyfillDvh_(size, pageId) {
     const {height, width} = size;
     if (height === 0 && width === 0) {
       return;
     }
-    this.storeService_.dispatch(Action.SET_PAGE_SIZE, {height, width});
     setImportantStyles(this.win.document.documentElement, {
       '--story-dvh': px(height / 100),
     });
+    if (pageId) {
+      const page = this.element.querySelector(
+        `amp-story-page#${escapeCssSelectorIdent(pageId)}`
```

Comment:
You could have the selector `amp-story-page[active]` here, and not require the parameter `pageId`